### ISSUE TITLE
Wrong colors at the edges of the screen

### ISSUE THREAD

**Issue by ehsan-sotoodeh** [pa7/heatmap.js#293]
At the edge of the screen, It shows a color that can not be true. for example, in blow screenshots, edges of the screen are blue but when we zoom out we see that it should be yellow. 

![2018-05-15 15_40_39-my map](https://user-images.githubusercontent.com/26613010/40085261-171fe612-5857-11e8-9d45-1bed8f08d150.png)

![2018-05-15 15_41_04-my map](https://user-images.githubusercontent.com/26613010/40085262-173b91c8-5857-11e8-8f06-e48bc7b31fe8.png)

![2018-05-15 15_42_40-my map](https://user-images.githubusercontent.com/26613010/40085264-17690a90-5857-11e8-8981-1de42e116ea3.png) 

![2018-05-15 15_41_40-v1 pptx - powerpoint](https://user-images.githubusercontent.com/26613010/40085263-175297d8-5857-11e8-83d3-35255f1d6542.png)

**Comment by jasperfirecai2** [comment]
Do you have `useLocalExtrema` set to true? this makes it compare data based on what on the screen right now which affects the color of spots

**Comment by ehsan-sotoodeh** [comment]
It's already false.

**Comment by jasperfirecai2** [comment]
it's possibly because datapoints get unloaded if their **center** is out of view, which can cause the data at high zoom levels to not include points out of view. I have the same "issue" but it's intended behaviour. if you need it to be accurate you're gonna need more data.

**Comment by ehsan-sotoodeh** [comment]
Thanks @jasperfirecai2.  I have a value for each half degree. In the images given above, there are hundreds of data points and I still get blue edges. 
you can check it here: http://gwf-demo.usask.ca/fifthversion/

**Comment by jasperfirecai2** [comment]
right. i know what the problem is, i just don't know how to solve it. when points overlap, their heatmap "value" increases, at the edges, there's less overlap happening because off-screen points don't get loaded at all. concentration doesn't really make a difference here. this is a tough one to solve.

**Comment by Woracheth** [comment]
You might try removing this code from gmaps-heatmap.js in plugins
```
      // we don't wanna render points that are not even on the map ;-)
      if (!bounds.contains(latlng)) {
        continue;
      }
```
If it works, the bounds variable is Google Maps LatLngBounds class. You might have to expand the bound.

**Comment by jcn-ithaca** [comment]
Seems like the "render bounds" should be larger than the "view bounds" with a margin that is equal to the radius, so that all points whose radii overlap the view get rendered

**Comment by UweKrause** [comment]
I know that this repository is not active, but for future reference I want to add an answer. Maybe it helps someone.

I made the same observation: Data points out of view are ignored (as intended).
But when they are just a little bit out of view, they should still have an influence to the heatmap.

I found a solution from user `ghybs` on stackoverflow:
https://stackoverflow.com/questions/34613289/disappearing-data-points-in-heatmap-js-leaflet-plugin

The solution is to add a padding to the bounds:
https://github.com/pa7/heatmap.js/blob/4e64f5ae5754c84fea363f0fcf24bea4795405ff/plugins/leaflet-heatmap/leaflet-heatmap.js#L101
```javascript
bounds = this._map.getBounds().pad(0.5);
```

`ghybs` used `0.5` in his answer, I have chosen the value of `0.75` by trial & error.
Try what works for you/your use case.
